### ISSUE TITLE
build error when CFG_CORE_MBEDTLS_MPI=n

### ISSUE THREAD

**Issue by vchong** [OP-TEE/optee_os#3487]
Hi maintainers,

With https://github.com/OP-TEE/optee_os/commit/47645577c806e80b6477193eff5ebc7de710ddf2 ("core: crypto: add support for SM3"), there's a build error when `CFG_CORE_MBEDTLS_MPI=n`.

```
core/crypto/sm3-hmac.c:12:10: fatal error: mbedtls/md.h: No such file or directory
```

**Comment by jforissier** [comment]
Hi Victor,

Oops! The fix is simple, just remove `#include <mbedtls/md.h>` in `core/crypto/sm3-hmac.c`. It is not used. I guess I mixed a few files when I was implementing SM3 and SM4 :-/

Perhaps adding `CFG_CORE_MBEDTLS_MPI=n` to Shippable would be a good idea.

**Comment by vchong** [comment]
Ok, I'll submit a PR. Thanks!

**Comment by vchong** [comment]
Fixed in #3488.
